opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/_view_instrument_match.py

```diff
@@ -124,46 +114,15 @@ def consume_measurement(self, measurement: Measurement) -> None:
 
         self._attributes_aggregation[attributes].aggregate(measurement)
 
-    def collect(self) -> Iterable[Metric]:
+    def collect(
+        self, aggregation_temporality: AggregationTemporality
+    ) -> Iterable[DataPointT]:
 
+        data_points = []
         with self._lock:
-            for (
-                attributes,
-                aggregation,
-            ) in self._attributes_aggregation.items():
-
-                previous_point = self._attributes_previous_point.get(
-                    attributes
-                )
-
-                current_point = aggregation.collect()
-
-                # pylint: disable=assignment-from-none
-
-                self._attributes_previous_point[
-                    attributes
-                ] = _convert_aggregation_temporality(
-                    previous_point,
-                    current_point,
-                    AggregationTemporality.CUMULATIVE,
+            for aggregation in self._attributes_aggregation.values():
+                data_points.append(
+                    aggregation.collect(aggregation_temporality)
```

Comment:
> Hmm quite a change in logic here.

Actually, there has been no logic change :slightly_smiling_face: 

> Why does this not yield now?

We used to use `yield` to generate a sequence of `Metric` objects by using it together with [`extend`](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/metric_reader_storage.py#L123). This is pretty much what we were doing:

``` python
def collect():
    for i in range(4):
        yield i


metrics = []

metrics.extend(collect())

print(metrics)
```
```
[0, 1, 2, 3]
```
We are now just returning a list. There are performance reasons to prefer a `yield`-generated list (by using a generator) when such list may not be iterated completely (because if you use a generator you don't end up generating all the elements at once, but you generate them as they are required, saving you from generating elements if the resulting list is not iterated completely). In this case we do use the whole list so there is not much value in using a generator.

> What does this change mean for the points that were being stored based on the unique label set which is no longer relevant?

It actually doesn't change, the logic has just been moved into the aggregations. We used to have a dictionary that kept the labels associated with the previous point. The labels are associated 1:1 to aggregations, so the keeping of the previous point has been moved to the [aggregations](https://github.com/open-telemetry/opentelemetry-python/pull/2658/files#diff-236d7fba7ea420c2160a5dfe4bb106c65e2b8160952e44c246d4c046c6e3dda9R68) (along with the rest of the `_convert_aggregation_temporality` logic), saving us from having an additional data structure.